lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1820,9 +1826,10 @@ estimateTxSize era skeleton =
 
     -- Total number of signatures the scripts require
     numberOf_ScriptVkeyWitnesses
-        = sumVia scriptRequiredKeySigs txScripts
+        = sumVia scriptRequiredKeySigs txMintOrBurnScripts
+        + maybe 0 scriptRequiredKeySigs txPaymentTemplate
```

Comment:
The effect of this line seems to be to: "add `sizeOf_VKeyWitness * scriptRequiredKeySigs` if a payment template exists." 

You also seem to be adding `determinePaymentTemplateSize` _once_.

I'd guess it should rather be `determinePaymentTemplateSize * nInputs * scriptRequiredKeySigs`?

Reply:
indeed! Thanks Johannes!